File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/sequrity/LoginUserCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.sequrity;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.DtoEntityInitialValueInterface;

/**
 * ユーザログインDto
 */
public class LoginUserCapsuleDto // NOPMD DataClass
        implements Serializable, DtoEntityInitialValueInterface {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** ユーザId */
    private String userId = INIT_STRING;

    /** パスワード */
    private String password = INIT_STRING;

    /**
     * ユーザIdを取得する
     *
     * @return ユーザId
     */
    public String getUserId() {
        return userId;
    }

    /**
     * ユーザIdを設定する
     *
     * @param userId ユーザId
     */
    public void setUserId(final String userId) {
        this.userId = userId;
    }

    /**
     * パスワードを取得する
     * 
     *
     * @return パスワード
     */
    public String getPassword() {
        return password;
    }

    /**
     * パスワードを設定する
     *
     * @param password パスワード
     */
    public void setPassword(final String password) {
        this.password = password;
    }

}

File: app/front/src/components/main/dto/login/loginUserCapsuleDto.ts
interface LoginUserCapsuleDtoInterface {

    /** ユーザId  */
    userId: string;

    /** パスワード */
    password: string;

    /** ログイン情報記憶 */
    rememberMe: boolean;
}

class LoginUserCapsuleDto implements LoginUserCapsuleDtoInterface {

    /** ユーザId  */
    userId: string;

    /** パスワード */
    password: string;

    /** ログイン情報記憶 */
    rememberMe: boolean;

    constructor() {
        const INIT_STRING: string = "";
        const INIT_BOOLEAN: boolean = false;

        this.userId = INIT_STRING;
        this.password = INIT_STRING;
        this.rememberMe = INIT_BOOLEAN;
    }
}

export { type LoginUserCapsuleDtoInterface, LoginUserCapsuleDto }
